### PR TITLE
Add ConsoleFormatter for Debugger::exportVar()

### PR DESCRIPTION
**Pull request by markstory**
This formatter adds ANSI escape code formatted output for use in console environments. I've not gone deep into 256 color land as I wanted to maximize compatibility. This also adds methods to Debugger to autoselect the formatter or users will be able to use `Debugger::configInstance()` to choose their preferred output format.

I chose to mostly duplicate the code in `TextFormatter` as I didn't want to clutter that class up with no-op `style()` calls. I'm also not confident in what a good abstraction would be or if its even useful. I hope to know more once I've implemented the HTML formatter.

### CODE REVIEW

File: src/Error/Debug/ConsoleFormatter.php [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+     *
+     * @param string $style The style name to use.
+     * @param string $text The text to style.
+     * @return string The styled output.
+     */
+    protected function style(string $style, string $text): string
+    {
+        $code = $this->styles[$style];
+
+        return "\033[{$code};m{$text}\033[0m";
```
Comment (ndm2) [inline src/Error/Debug/ConsoleFormatter.php]: I'm not sure, but is that closing semicolon there actually valid?

```suggestion
        return "\033[{$code}m{$text}\033[0m";
```

File: src/Error/Debug/ConsoleFormatter.php [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+     *
+     * @param \Cake\Error\Debug\ArrayNode $var The array to export.
+     * @param int $indent The current indentation level.
+     * @return string Exported array.
+     */
+    protected function _array(ArrayNode $var, int $indent): string
+    {
+        $out = $this->style('punct', '[');
+        $break = "\n" . str_repeat("  ", $indent);
+        $end = "\n" . str_repeat("  ", $indent - 1);
```
Comment (ndm2) [inline src/Error/Debug/ConsoleFormatter.php]: Could we get 4 spaces of indentation?

File: src/Error/Debug/ConsoleFormatter.php [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+        }
+
+        $out = $this->style('punct', 'object(') .
+            $this->style('class', $var->getValue()) .
+            $this->style('punct', ') id:') .
+            $this->style('number', (string)$var->getId()) .
+            $this->style('punct', ' {');
+
+        $break = "\n" . str_repeat("  ", $indent);
+        $end = "\n" . str_repeat("  ", $indent - 1) . $this->style('punct', '}');
```
Comment (ndm2) [inline src/Error/Debug/ConsoleFormatter.php]: Could we get 4 spaces of indentation?

File: src/Error/Debug/ConsoleFormatter.php (hunk 2 above)
Comment (markstory) [reply src/Error/Debug/ConsoleFormatter.php]: I went to 2 spaces to try and keep long content in narrow terminals better. I can go back to 4 though.

File: src/Error/Debug/ConsoleFormatter.php (hunk 1 above)
Comment (markstory) [reply src/Error/Debug/ConsoleFormatter.php]: It seems to work for me in iTerm with and without the `;` so we can do without.

File: src/Error/Debug/ConsoleFormatter.php (hunk 2 above)
Comment (ndm2) [reply src/Error/Debug/ConsoleFormatter.php]: That's a good point... given that everything's consistently formatted, 2 spaces aren't too bad :)

**Pull request merged**